File: server/Entities/MenuItem.cs
namespace server.Entities;

public class MenuItem : BaseEntity
{
    public required string Name { get; set; } = "";
    public decimal Price { get; set; }
    public string? Description { get; set; } = "";
    public required string ImageUrl { get; set; } = "";
    public required string Category { get; set; } = "";
    public bool IsAvailable { get; set; } = true;
    // UI flags
    public bool IsBest { get; set; } = false;
    public bool IsVeg { get; set; } = false;
    public bool IsSpicy { get; set; } = false;
}

File: client/Restaurant-Dashboard/types/menuItem.ts
export interface MenuItem {
    id: number
    name: string
    price: number
    description?: string
    imageUrl?: string
    category?: string
    isAvailable: boolean
    isBest?: boolean
    isVeg?: boolean
    isSpicy?: boolean
    createdAt?: string
    updatedAt?: string
}

export type MenuItemPayload = Omit<MenuItem, "id" | "createdAt" | "updatedAt">